fdbcli/SetClassCommand.actor.cpp

```diff
@@ -73,6 +73,24 @@ ACTOR Future<Void> printProcessClass(Reference<IDatabase> db) {
 };
 
 ACTOR Future<bool> setProcessClass(Reference<IDatabase> db, KeyRef network_address, KeyRef class_type) {
+	// Check if the network address filter matches any processes before proceeding.
+	state Reference<ITransaction> preCheckFilterTr = db->createTransaction();
+	loop {
+		preCheckFilterTr->setOption(FDBTransactionOptions::SPECIAL_KEY_SPACE_ENABLE_WRITES);
```

Comment:
I don't think you'll need this option if you split them into two transactions since this one is doing a read, but I think I agree with your thought that the most elegant solution would be to combine the two transactions into one.

I also notice that the second transaction specifically checks for a special keys API failure. I'm not sure if that's needed in this transaction, though combining into one transaction makes that question moot.

Reply:
Updated in `662ff40`